src/tests/TestCrypt/TestCrypt.cpp

```diff
@@ -251,6 +251,22 @@ void TestCrypt::xexstarAttack() {
 	// Make sure we detected the attack
 	QVERIFY(failed_encrypt);
 	QVERIFY(failed_decrypt);
+
+	// The assumption that critical packets do not turn up by pure chance turned out to be incorrect
+	// since digital silence appears to produce them in mass.
+	// So instead we now modify the packet in a way which should not affect the audio but will
+	// prevent the attack.
+	QVERIFY(cs.ocb_encrypt(src, encrypted, 2 * AES_BLOCK_SIZE, nonce, enctag));
+	QVERIFY(cs.ocb_decrypt(encrypted, decrypted, 2 * AES_BLOCK_SIZE, nonce, dectag));
+
+	// Tags should match
+	for (int i = 0; i < AES_BLOCK_SIZE; ++i) {
+		QCOMPARE(enctag[i], dectag[i]);
+	}
+
+	// Actual content should have been change such that the critical block is no longer all 0.
+	QCOMPARE(src[0], 0);
+	QCOMPARE(decrypted[0], 1);
```

Comment:
```suggestion
	QCOMPARE(src[0], static_cast<unsigned char>(0));
	QCOMPARE(decrypted[0], static_cast<unsigned char>(1));
```

Should fix the compile issue on Linux (the arguments to `QCOMPARE` are required to have the same type).
```cpp
/home/vsts/work/1/s/src/tests/TestCrypt/TestCrypt.cpp:268: undefined reference to `bool QTest::qCompare<unsigned char, int>(unsigned char const&, int const&, char const*, char const*, char const*, int)'
/home/vsts/work/1/s/src/tests/TestCrypt/TestCrypt.cpp:269: undefined reference to `bool QTest::qCompare<unsigned char, int>(unsigned char const&, int const&, char const*, char const*, char const*, int)'
```